### ISSUE TITLE
Elastic throws 403 (read-only index), but Bulk.Do(ctx) does not catch an error

### ISSUE THREAD

**Issue by mshirman** [olivere/elastic#1277]
Please use the following questions as a guideline to help me answer
your issue/question without further inquiry. Thank you.

### Which version of Elastic are you using?

[ ] elastic.v7 (for Elasticsearch 7.x)
[x ] elastic.v6 (for Elasticsearch 6.x)
[ ] elastic.v5 (for Elasticsearch 5.x)
[ ] elastic.v3 (for Elasticsearch 2.x)
[ ] elastic.v2 (for Elasticsearch 1.x)

### Please describe the expected behavior
Expect Bulk.Do() to return an error when Elastic throws 403 upon at attempt to do bulk indexing into a  read-only index.

` 403 ({"type"=>"cluster_block_exception", "reason"=>"blocked by: [FORBIDDEN/12/index read-only / allow delete (api)]`

### Please describe the actual behavior
My indexes were in the read-only state, bulk indexing was seemingly completing without an issue. `Bulk.Do(ctx)` was not returning an error.

I troubleshoot by `put` a doc manually and received the 403 (shown above).

### Any steps to reproduce the behavior?
I assume you should be able to set an index to a read-only state and attempt to bulk index into the index...

`PUT /<indexname>/_settings
{
"index": {
     "blocks": {
          "read_only_allow_delete": "true"
      }
   }
}`

**Comment by olivere** [comment]
This client should simply route through all errors returned from ES. So if ES returns an error, the client should. And the client should not throw any errors if ES doesn't, otherwise we don't have consistent behavior.

Can you confirm, by a bunch of `curl` commands, that ES _does_ return an error when doing the above?

**Comment by olivere** [comment]
Does ES perhaps return errors, but on the item-level? [This is the response structure](https://github.com/olivere/elastic/blob/release-branch.v7/bulk.go#L323-L368).

**Comment by mshirman** [comment]
Thank you for the response. I will try to get back to this and test. Will let you know.

**Comment by ikingme** [comment]
This is a problem about the BULK api.   In another issue  there are more details. https://github.com/olivere/elastic/issues/1258

**Comment by olivere** [comment]
I added a [test](76080bbe03f9f204998f1a6f582b7764e8695efe) against an ES 7.x cluster. This is by design and works as expected. The request/response looks like this:

```
POST /_bulk?pretty=true HTTP/1.1
Host: 127.0.0.1:9200
User-Agent: elastic/7.0.17 (darwin-amd64)
Content-Length: 160
Accept: application/json
Content-Type: application/x-ndjson
Accept-Encoding: gzip

{"index":{"_index":"elastic-test","_id":"1"}}
{"user":"olivere","message":"Welcome to Golang and Elasticsearch.","retweets":0,"created":"0001-01-01T00:00:00Z"}

HTTP/1.1 200 OK
Content-Type: application/json; charset=UTF-8

{
  "took" : 0,
  "errors" : true,
  "items" : [
    {
      "index" : {
        "_index" : "elastic-test",
        "_type" : "_doc",
        "_id" : "1",
        "status" : 429,
        "error" : {
          "type" : "cluster_block_exception",
          "reason" : "index [elastic-test] blocked by: [TOO_MANY_REQUESTS/12/index read-only / allow delete (api)];"
        }
      }
    }
  ]
}
```

As you can see, ES responds with HTTP Status 200 (OK), which indicates that the Bulk API call was successful. However, the individual document you wanted to bulk index has failed, as you can see with the individual `"status": 429`. Also, there's a general `"errors": true` flag in the response indicating if _any_ bulk request has failed.

This is good, in general, as it allows you to send many requests to different indices, and the Bulk API succeeds even if some documents fail. You as a caller need to figure out what happened.
